django/middleware/csrf.py

```diff
@@ -63,19 +99,37 @@ def rotate_token(request):
     """
     request.META.update({
         "CSRF_COOKIE_USED": True,
-        "CSRF_COOKIE": _get_new_csrf_key(),
+        "CSRF_COOKIE": _get_new_csrf_token(),
     })
+    request.csrf_cookie_needs_reset = True
 
 
 def _sanitize_token(token):
     # Allow only alphanum
-    if len(token) > CSRF_KEY_LENGTH:
-        return _get_new_csrf_key()
-    token = re.sub('[^a-zA-Z0-9]+', '', force_text(token))
-    if token == "":
-        # In case the cookie has been truncated to nothing at some point.
-        return _get_new_csrf_key()
-    return token
+    if re.search('[^a-zA-Z0-9]', force_text(token)):
+        return None
+    elif len(token) == CSRF_TOKEN_LENGTH:
+        return token
+    elif len(token) == CSRF_NONCE_LENGTH:
+        # Older Django versions set cookies to values of CSRF_NONCE_LENGTH
+        # alphanumeric characters. For backwards compatibility,  we accept
+        # such values as unpadded nonces.
+        # It's easier to pad here and be consistent later, rather than add
+        # different code paths in the checks, although that might be a tad more
+        # efficient.
+        return _pad_cipher_nonce(token)
+    return None
+
+
+def _compare_padded_tokens(request_csrf_token, csrf_token):
+    if not(request_csrf_token and csrf_token):
```

Comment:
Is that check actually needed? This seems to be only needed if a token is `None` -- which would not happen if we return a token from `sanitize_token` instead of None :D